proj/src/main/webapp/script.js

```diff
@@ -1,24 +1,32 @@
+const firebase = require('firebase');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+window = (new JSDOM('')).window;
+document = window.document
+const $ = require('jquery');
+require('bootstrap'); // for collapse
+
 /*
     Notifications
  */
 
-const messaging = firebase.messaging();
-messaging.requestPermission()
-.then(function () {
-    console.log("Have permission");
-    if ('serviceWorker' in navigator) {
-        navigator.serviceWorker.register('/firebase-messaging-sw.js')
-        .then(function(registration) {
-            console.log('Registration successful, scope is:', registration.scope);
-        }).catch(function(err) {
-            console.log('Service worker registration failed, error:', err);
-        });
-    }
-});
-
-messaging.onMessage((payload) => {
-    appendMessage(payload);
-});
+// const messaging = firebase.messaging();
+// messaging.requestPermission()
+// .then(function () {
+//     console.log("Have permission");
+//     if ('serviceWorker' in navigator) {
+//         navigator.serviceWorker.register('/firebase-messaging-sw.js')
+//         .then(function(registration) {
+//             console.log('Registration successful, scope is:', registration.scope);
+//         }).catch(function(err) {
+//             console.log('Service worker registration failed, error:', err);
+//         });
+//     }
+// });
+
+// messaging.onMessage((payload) => {
+//     appendMessage(payload);
+// });
```

Comment:
Does this still need to be here?

Reply:
I'm not sure. It was failing and breaking the build because of these changes, but according to nlumiere, they were not functional before, but just failing silently. So I just commented it out so that he can return to it and fix it up later.